dockerfiles/lib/src/api/wsmaster/permissions/permissions.ts

```diff
@@ -36,9 +42,9 @@ export class Permissions {
      */
     listPermissions():Promise<Array<DomainDto>> {
 
-        var jsonRequest:HttpJsonRequest = new DefaultHttpJsonRequest(this.authData, '/api/permissions', 200);
+        let jsonRequest: HttpJsonRequest = new DefaultHttpJsonRequest(this.authData, this.apiLocation, '/api/permissions', 200);
```

Comment:
why do we need to provide apiLocation if authData contains the server location ?

Reply:
Because keycloak is deployed outside of Che API

Reply:
ok I think the main issue comes that global idea was to only provide one object to the instances including all the configuration. So maybe we could have merged one object into another one (or a third one)

so, we could have all auth the authentication data stored in a single object along with the api location

Reply:
In case of exec agent of the wsagent location of the API that is being requested has no connection with authentication at all. Do you think we should keep this in the auth-data?
If so then it doesn't look like auth data only and this behavior is not described in the docs of this class.
I like simplicity too, but in this case, I don't see how server of wsagent fits in auth-data class.
If I got something wrong, please, elaborate.

Reply:
What if we do next:
1. Allow getting API location from AuthData
2. Make HttpJsonRequest constructor accept AuthData and ServerLocation
3. In HttpJsonRequest if provided ServerLocation is null then use server location from AuthData

Or we can alternate 2 and 3 with the addition of a method to HttpJsonRequest which sets ServerLocation for the request. If method is not called ServerLocation from AuthData is used.

Reply:
I like this proposal